test/integration/diag_nodes_test.go

```diff
@@ -55,26 +60,32 @@ func TestDiagNodeConditions(t *testing.T) {
 	}
 
 	// Make the node unschedulable and verify diagnostics notices
-	nodeList.Items[0].Spec.Unschedulable = true
-	if _, err := client.Nodes().Update(&(nodeList.Items[0])); err != nil {
-		t.Fatalf("expected no errors making node unschedulable, but: %#v", err)
-	}
-	result = nodeDiag.Check()
-	if errors := result.Errors(); len(errors) != 1 ||
-		!diagtype.MatchesDiagError(errors[0], "DClu0004") {
-		t.Fatalf("expected 1 error about not having nodes, but: %#v", errors)
-	} else if warnings := result.Warnings(); len(warnings) < 1 || !diagtype.MatchesDiagError(warnings[0], "DClu0003") {
-		t.Fatalf("expected a warning about test-node not being schedulable, but: %#v", warnings)
-	}
+	waitForNode(t, client, 3, "making node unschedulable", func(nodeList *kapi.NodeList, t *testing.T) (bool, error) {
+		nodeList.Items[0].Spec.Unschedulable = true
+		if _, err := client.Nodes().Update(&(nodeList.Items[0])); err != nil {
+			return false, err
+		}
+		result = nodeDiag.Check()
```

Comment:
the `waitForNode` handler should just be doing the node update, move the `nodeDiag.Check()` stuff out